### ISSUE TITLE
[FrameworkBundle] URL for setting the request context

### ISSUE THREAD

**Issue by alanpoulain** [symfony/symfony#35121]
**Description**  
Today when we want to change the generated URLs in commands, we need to set a lot of parameters (scheme, host, port, ...): https://symfony.com/doc/current/routing.html#generating-urls-in-commands.

It could be useful and less cumbersome to set an URL instead. It would be parsed in a compiler pass and the parameters would be set there.

**Example**  
Before:
```yaml
# config/services.yaml
parameters:
    router.request_context.scheme: 'https'
    router.request_context.host: 'example.org'
    request_listener.https_port: 8443
```

After:
```yaml
# config/services.yaml
parameters:
    router.request_context.url: 'https://example.org:8443'
```

**Comment by pyatnitsev** [comment]
I think, I can try to do that. I think this case can be solved next way:

1. Create new CompillerPass, that check new parameter 
2. Parse url and put parsed parameters into old vars 

WDYT?

**Comment by nicolas-grekas** [comment]
Implemented in #36651
